tcases-lib/src/main/java/org/cornutum/tcases/io/ObjectUtils.java

```diff
@@ -0,0 +1,144 @@
+//////////////////////////////////////////////////////////////////////////////
+// 
+//                    Copyright 2018, Cornutum Project
+//                             www.cornutum.org
+//
+//////////////////////////////////////////////////////////////////////////////
+
+package org.cornutum.tcases.io;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * Defines utility methods for reading and writing objects.
+ *
+ */
+public final class ObjectUtils
+  {
+  private ObjectUtils()
+    {
+    // Static methods only
+    }
+  
+  /**
+   * Returns the object represented by the given string.
+   */
+  public static Object toObject( String value)
+    {
+    return toObject( value, toBoolean, toNumber, toString);
+    }
+  
+  
+  /**
+   * Returns the first non-null object produced by the given converters.
+   * Returns null if no such object found.
+   */
+  @SafeVarargs
+  private static <T> Object toObject( T value, Function<T,Object>... converters)
+    {
+    return
+      Stream.of( converters)
+      .map( converter -> converter.apply( value))
+      .filter( Objects::nonNull)
+      .findFirst()
+      .orElse( null);
+    }
+
+  /**
+   * Returns the Boolean represented by the given string, or null if not a boolean value.
+   */
+  private static Function<String,Object> toBoolean =
+    value ->
+      "true".equalsIgnoreCase( value)?
+      Boolean.TRUE :
+
+      "false".equalsIgnoreCase( value)?
+      Boolean.FALSE :
+
+      null;
+
+  /**
+   * Returns the given String value.
+   */
+  private static Function<String,Object> toString =
```

Comment:
hehe, actually you can just use static methods and method references. Why didn't I see this yesterday?

Reply:
Brilliant!